sys/include/ztimer64.h

```diff
@@ -0,0 +1,514 @@
+
+/**
+ * @brief   Post a message after a delay (relative version)
+ *
+ * This function sets a timer that will send a message @p offset ticks
+ * from now.
+ *
+ * @note The memory pointed to by @p timer and @p msg will not be copied, i.e.
+ *       `*timer` and `*msg` needs to remain valid until the timer has triggered.
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[in]   timer           ztimer timer struct to use
+ * @param[in]   target          target time
+ * @param[in]   msg             pointer to msg that will be sent
+ * @param[in]   target_pid      pid the message will be sent to
+ */
+static inline void ztimer64_set_msg(ztimer64_clock_t *clock, ztimer64_t *timer,
+                                    uint32_t offset,
+                                    msg_t *msg, kernel_pid_t target_pid)
+{
+    ztimer64_set_msg_at(clock, timer, ztimer64_offset2absolute(clock,
+                                                               offset), msg,
+                        target_pid);
+}
+
+/**
+ * @brief receive a message (blocking, with absolute timeout)
+ *
+ * Similar to msg_receive(), but with a timeout parameter.
+ * The function will return after waiting at most until @p target.
+ *
+ * @note: This might function might leave a message with type MSG_ZTIMER64 in the
+ *        thread's message queue, which must be handled (ignored).
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[out]  msg             pointer to buffer which will be filled if a
+ *                              message is received
+ * @param[in]   target          absolute target, in @p clock time
+ *
+ * @return  >=0 if a message was received
+ * @return  -ETIME on timeout
+ */
+int ztimer64_msg_receive_until(ztimer64_clock_t *clock, msg_t *msg,
+                               uint64_t target);
+
+/**
+ * @brief receive a message (blocking, with relative timeout)
+ *
+ * Similar to msg_receive(), but with a timeout parameter.
+ * The function will return after waiting at most @p timeout ticks.
+ *
+ * @note: This might function might leave a message with type MSG_ZTIMER64 in the
+ *        thread's message queue, which must be handled (ignored).
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[out]  msg             pointer to buffer which will be filled if a
+ *                              message is received
+ * @param[in]   timeout         relative timeout, in @p clock time units
+ *
+ * @return  >=0 if a message was received
+ * @return  -ETIME on timeout
+ */
+static inline int ztimer64_msg_receive_timeout(ztimer64_clock_t *clock,
+                                               msg_t *msg,
+                                               uint32_t timeout)
+{
+    return ztimer64_msg_receive_until(clock, msg,
+                                      ztimer64_offset2absolute(clock, timeout));
+}
+
+#define MSG_ZTIMER64 0xc83f   /**< msg type used by ztimer64_msg_receive_timeout */
```

Comment:
Chosen by fair dice roll?
![xkcd](https://imgs.xkcd.com/comics/random_number.png)

Image embedded from https://xkcd.com/221/ - all credits to xkcd.

Reply:
hah! someone noticed. `MSG_ZTIMER` was created with the define generator that's PR'ed somewhere.
I just used that value, changed a bit, and removed the comment.